server/project-service/src/main/java/edu/illinois/ncsa/incore/service/project/controllers/ProjectController.java

```diff
@@ -120,12 +121,41 @@ public List<Project> getProjectsList(@Parameter(name = "Name filter") @QueryPara
             queryMap.put("region", region);
         }
 
-        List<Project> projects;
+        List<Project> projects = new ArrayList<>();
 
-        if (queryMap.isEmpty()) {
+        if (searchText != null && !searchText.isEmpty()) {
+            Project project = this.projectDAO.getProjectById(searchText);
+            if (project != null) {
+                projects.add(project);
+            } else {
+                projects = this.projectDAO.searchProjects(searchText);
```

Comment:
what if this projects is also null?

Reply:
I think the search function will always return list, if not found then empty list.
```
@Override
    public List<Project> searchProjects(String text) {
        Query<Project> query = this.dataStore.find(Project.class).filter(
            Filters.or(
                Filters.regex(PROJECT_FIELD_NAME).pattern(text).caseInsensitive(),
                Filters.regex(PROJECT_FIELD_DESCRIPTION).pattern(text).caseInsensitive(),
                Filters.regex(PROJECT_FIELD_CREATOR).pattern(text).caseInsensitive(),
                Filters.regex(PROJECT_FIELD_OWNER).pattern(text).caseInsensitive(),
                Filters.regex(PROJECT_FIELD_REGION).pattern(text).caseInsensitive()
            ));
        return query.iterator().toList();
    }
    ```